annotation-utils/src/main/java/edu/isi/vista/annotationutils/Rehydrate.kt

```diff
@@ -0,0 +1,75 @@
+package edu.isi.vista.annotationutils
+
+import edu.isi.nlp.io.OffsetIndexedCorpus
+import edu.isi.nlp.symbols.Symbol
+import edu.isi.nlp.io.DocIDToFileMappings
+import com.google.common.base.Optional
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.isi.nlp.parameters.Parameters
+import java.io.File
+
+/**
+ * Puts the document text back into the 'sofaString' field inside json files created by `ExportAnnotations.kt`
+ *
+ * The program takes one argument, a parameter file in the format described at
+ * https://github.com/isi-vista/nlp-util/blob/master/common-core-open/src/main/java/edu/isi/nlp/parameters/serifstyle/SerifStyleParameterFileLoader.java
+ *
+ * An example is provided in curated-training-annotator/sample_params/rehydrate.params
+ * The required parameter values are:
+ * <ul>
+ *     <li> {@code indexDirectory} is the location of the files produced by {@code IndexFlatGigaword.java}
+            (https://github.com/isi-vista/nlp-util/blob/master/nlp-core-open/src/main/java/edu/isi/nlp/corpora/gigaword/IndexFlatGigaword.java) </li>
+ *     <li> {@code gigawordDataDirectory} is the location of the gigaword text files </li>
+ *     <li> {@code jsonInputDirectory} is the location of the 'dehydrated' json files produced by {@code ExportAnnotations.kt} </li>
+ *     <li> {@code rehydratedJsonDirectory} is where the new json files will go </li>
+ *  </ul>
+ */
+
+fun main(argv: Array<String>) {
+    // Load parameters:
+    val params = Parameters.loadSerifStyle(File(argv[0]))
+    val indexDirectory = params.getExistingDirectory("indexDirectory")
+    val gigawordDataDirectory = params.getExistingDirectory("gigawordDataDirectory")
+    val jsonInputDirectory = params.getCreatableDirectory("dehydratedJsonDirectory")
+    val outputDirectory = params.getCreatableDirectory("rehydratedJsonDirectory")
+    outputDirectory.toString()
+
+    // Create an OriginalTextSource for getting original document text to put in json:
+    val textMap = DocIDToFileMappings.forFunction { symbol ->
```

Comment:
You could consider pulling everything from here through lines 44 into its own method (possibly just taking the `Parameters` as an argument)